modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4TcpChannel.java

```diff
@@ -72,7 +72,9 @@ public void addCloseListener(ActionListener<Void> listener) {
 
     @Override
     public void setSoLinger(int value) {
-        channel.config().setOption(ChannelOption.SO_LINGER, value);
+        if (channel.isOpen()) {
+            channel.config().setOption(ChannelOption.SO_LINGER, value);
```

Comment:
I guess this could still throw on us? How about we catch the channel exception here and re-throw it as an I/O exception? Then I think we are covered on all bases.